src/test/java/uk/gov/hmcts/reform/professionalapi/util/RefDataUtilTest.java

```diff
@@ -407,5 +417,88 @@ public void test_updateUserDetailsForActiveOrganisation() {
+        professionalUsersResponse1.setIdamStatus(IdamStatus.ACTIVE.toString());
+        professionalUsersResponse2.setIdamStatus(IdamStatus.PENDING.toString());
+
+        List<ProfessionalUsersResponse> userProfiles = asList(professionalUsersResponse, professionalUsersResponse1, professionalUsersResponse2);
+        ProfessionalUsersEntityResponse professionalUsersEntityResponse = new ProfessionalUsersEntityResponse();
+        professionalUsersEntityResponse.setUserProfiles(userProfiles);
+
+        HttpHeaders header = new HttpHeaders();
+        header.setContentType(APPLICATION_JSON);
+        List<SuperUser> users = new ArrayList<>();
+        users.add(professionalUser.toSuperUser());
+        organisation.setUsers(users);
+        Map<String, Organisation> activeOrganisationDtls = new HashMap<>();
+        activeOrganisationDtls.put("1",organisation);
+        activeOrganisationDtls.put("2",organisation);
+        activeOrganisationDtls.put("3",organisation);
+        ResponseEntity<?> realResponseEntity = new ResponseEntity<>(professionalUsersEntityResponse, header, HttpStatus.OK);
+        Map<String, Organisation> response = RefDataUtil.updateUserDetailsForActiveOrganisation(realResponseEntity, activeOrganisationDtls);
+
+
+        SuperUser item = ((SuperUser)users.get(0));
+        assertNull(item.getId());
+        assertEquals("fName", item.getFirstName());
+        assertEquals("lName", item.getLastName());
+        assertEquals("some@email.com", item.getEmailAddress());
+        assertNull(item.getOrganisation().getId());
+        assertEquals("Org-Name", item.getOrganisation().getName());
+    }
+
+    @Test(expected = ExternalApiException.class)
+    public void testGetSingleUserIdFromUserProfile() throws Exception {
+        Map<String, Collection<String>> header = new HashMap<>();
+        Collection<String> list = new ArrayList<>();
+        header.put("content-encoding", list);
+        String body = "{"
+            + "  \"errorMessage\": \"400\","
+            + "  \"errorDescription\": \"BAD REQUEST\","
+            + "  \"timeStamp\": \"23:10\""
+            + "}";
+
+        Response response = Response.builder().status(400).reason("BAD REQUEST").headers(header).body(body, UTF_8).request(mock(Request.class)).build();
+        when(userProfileFeignClient.getUserProfileById(any())).thenReturn(response);
+
+        ProfessionalUser result = RefDataUtil.getSingleUserIdFromUserProfile(new ProfessionalUser("firstName", "lastName", "emailAddress", new Organisation("name", OrganisationStatus.PENDING, "sraId", "companyNumber", Boolean.TRUE, "companyUrl")), userProfileFeignClient, Boolean.TRUE);
+        assertEquals(new ProfessionalUser("firstName", "lastName", "emailAddress", new Organisation("name", OrganisationStatus.PENDING, "sraId", "companyNumber", Boolean.TRUE, "companyUrl")), result);
+        verify(userProfileFeignClient.getUserProfileById(any()),times(1));
```

Comment:
Please change this to `verify(userProfileFeignClient, times(1)).getUserProfileById(any());`

Reply:
Done